app/lib/api_error/saved_page_not_destroyed.rb

```diff
@@ -0,0 +1,21 @@
+module ApiError
+  class SavedPageNotDestroyed < ApiError::Base
+    def initialize(page_path:)
+      @page_path = page_path
+    end
+
+    def status_code
+      :unprocessable_entity
```

Comment:
I think this should be a 404: the saved page couldn't be destroyed because it's not found.

Reply:
There's [a separate 404 check](https://github.com/alphagov/account-api/pull/69/commits/6a35b1b0e57129feb86fabb8ce5d0ef02248be4f) that is applied as a before action to any route where a page needs to be identified (`show` and `destroy` currently)

This check is more a piece of defensive programming, I can't figure out how anyone would ever end up on this branch, but broadly:
- if they supply a good path
- and they are authenticated
- and yet when we call destroy on the model... it did not work

Then we show the saved page not destroyed error.

Both that and SavedPageNotPersisted shouldn't ever crop up, but if they do it'd be worth us checking those out in sentry!

Reply:
Aha, I see.  Won't `destroy!` raise an exception if something goes wrong though?  So this bit of defensive code will never be reached?

- If the page doesn't exist, we return a 404
- If it does, and we delete it, we return a 204
- If it does, and deleting it fails, an exception is thrown and we return a 500

Reply:
ah that's a great point!

Initially we had these all working by rescuing raised exceptions, but most of the force pushing above is us changing that to look at validations.

Perhaps I want to change this then to `destroy`, which i assume makes the attempt, populates the errors and does not raise if something is wrong...

Then put this right and the end and say if it's still `.persisted?` then... throw this error!

Reply:
or perhaps 500 and an error state is correct in this case 🤔 
I'm kind of putting the blame on the user with a 4xx error...

Reply:
I think a 500 is the more appropriate error here, so the controller method can just be

```ruby
  def destroy
    saved_page.destroy!
    head :no_content
  end
```

And this error class doesn't need to exist